gmso/core/virtual_site.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, List, Optional, Union
+
+import unyt as u
+from pydantic import Field
+
+from gmso.abc.abstract_site import Site
+from gmso.core.virtual_type import VirtualPositionType, VirtualPotentialType
+
+
+class VirtualSite(Site):
+    """A generalized virtual site class in GMSO.
+
+    Virtual sites are massless particles that represent off-atom charge sites, lone pairs, or other non-physical sites.
+
+    Attributes
+    ----------
+    charge : float
+        The charge of the virtual site in elementary charge units.
+    parent_atoms : List[Site]
+        The real constituent atoms that define the virtual site's position.
+    virtual_type:
```

Comment:
It looks like we still need to finish the doc strings here.

Reply:
Great catch. Fixed in 80f407a